istioctl/pkg/tag/generate.go

```diff
@@ -86,45 +89,33 @@ type GenerateOptions struct {
 	// UserManaged indicates whether the revision tag is user managed.
 	// If true, the revision tag will not be affected by the installer.
 	UserManaged bool
+	// IstioNamespace indicates the namespace of the istio installation.
+	IstioNamespace string
 }
 
 // Generate generates the manifests for a revision tag pointed the given revision.
-func Generate(ctx context.Context, client kube.Client, opts *GenerateOptions, istioNS string) (string, error) {
+func Generate(ctx context.Context, client kube.Client, opts *GenerateOptions) (string, error) {
 	// abort if there exists a revision with the target tag name
-	revWebhookCollisions, err := GetWebhooksWithRevision(ctx, client.Kube(), opts.Tag)
+	err := checkTagNameCollidesWithRevisionName(ctx, client.Kube(), opts)
```

Comment:
Do we also need to do this check when we create revisions?

Reply:
yes

Reply:
fwiw, today `istioctl tag set` can be used to convert a revision name to a tag name, which is something similar to what you're saying.

https://github.com/istio/istio/issues/56128

We could do it, but perhaps as part of another PR after gathering more info in that issue?